src/Libraries/PythonNodeModelsWpf/PythonNode.cs

```diff
@@ -157,6 +157,7 @@ private void EditScriptContent(object sender, EventArgs e)
         private void UpdateToPython2Engine(object sender, EventArgs e)
         {
             pythonNodeModel.Engine = PythonEngineVersion.IronPython2;
+            pythonNodeModel.OnNodeModified();
```

Comment:
Isn't this equivalent to calling `OnNodeModified` in the setter of the `Engine` property?

Reply:
That way 4 calls to `OnNodeModified` are reduced to one.

Reply:
yes, please see the conversation above, by having the engine property setter always call `OnNodeModified()` there is no way to avoid multiple runs which are triggered by setting the code content, since these also trigger executions - and it is difficult to ensure the engine run is actually executing with valid code (most updated).

This implementation has a bit more overhead but aligns `script` and `engine` setters, they now both require calling `NodeModified` and let's us control when to execute more precisely.

Reply:
@aparajit-pratap This is purposefully written to avoid multiple runs of the same python code.